### PR TITLE
Strip whitespaces from names before saving organisation, scheme or a user

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Saving it with a space was erroring out when trying to select them from an accessible select. 
It also makes sense not to store the leading/trailing spaces in the names

### CODE REVIEW

**Comment by baarkerlounger** [comment]
We probably want this on location name as well?

File: app/models/organisation.rb [hunk 1]
```diff
@@ -74,4 +76,8 @@ def display_attributes
       { name: "data_protection_agreement", value: data_protection_agreement_string, editable: false },
     ]
   end
+
+  def strip_whitespace
+    self.name = name.strip unless name.nil?
```
Comment (baarkerlounger) [inline app/models/organisation.rb]: https://api.rubyonrails.org/classes/String.html#method-i-squish might be better than strip?

File: app/models/scheme.rb [hunk 2]
```diff
@@ -228,4 +230,8 @@ def validate_confirmed
       end
     end
   end
+
+  def strip_whitespace
+    self.service_name = service_name.strip unless service_name.nil?
```
Comment (baarkerlounger) [inline app/models/scheme.rb]: I wonder if this is worth genericising to all fields that respond to strip/squish and extracting to `ApplicationRecord`?

File: app/models/scheme.rb [hunk 3]
```diff
@@ -19,6 +19,8 @@ class Scheme < ApplicationRecord
 
   validate :validate_confirmed
 
+  before_validation :strip_whitespace
```
Comment (baarkerlounger) [inline app/models/scheme.rb]: Could just override `field=` methods... (has the advantage of setting it in memory the same as you'll save it already before actually doing the save, but might be unnecessary indirection...)

File: app/models/organisation.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/models/organisation.rb]: Hmm, could there ever be a reason to have a double space in the name?

File: app/models/scheme.rb (hunk 2 above)
Comment (kosiakkatrina) [reply app/models/scheme.rb]: Yeah, possibly, will have a look

File: app/models/organisation.rb (hunk 1 above)
Comment (baarkerlounger) [reply app/models/organisation.rb]: Not that I know of, but I could be wrong.

Comment (kosiakkatrina) [reply app/models/organisation.rb]: If we're generalising this, I think I'd prefer just to strip it for now, cause it feels less destructive

File: app/models/scheme.rb (hunk 3 above)
Comment (kosiakkatrina) [reply app/models/scheme.rb]: How strongly do you feel about it?

File: app/models/application_record.rb [hunk 4]
```diff
@@ -1,3 +1,8 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  def strip_whitespaces
+    self.service_name = service_name.strip unless !respond_to?("service_name") || service_name.nil?
```
Comment (baarkerlounger) [inline app/models/application_record.rb]: Do you need the nil checks here? Wouldn't that be caught by the respond_to already?

Comment (baarkerlounger) [reply app/models/application_record.rb]: You could maybe also have this take (*args) that you iterate over and the models could pass the fields that need stripping?

Comment (kosiakkatrina) [reply app/models/application_record.rb]: can't pass in as a param from the before_validation, but added a `fields_to_strip` method that returns those fields for each class

File: Gemfile [hunk 5]
```diff
@@ -55,6 +55,7 @@ gem "redis"
 gem "sentry-rails"
 gem "sentry-ruby"
 # Possessives in strings
+gem "auto_strip_attributes"
```
Comment (baarkerlounger) [inline Gemfile]: Need to fix the comments here

**Review by baarkerlounger — APPROVED** [review]

**Pull request merged**